packages/stark-ui/src/modules/language-selector/components/language-selector.component.ts

```diff
@@ -0,0 +1,126 @@
+
+export type StarkLanguageSelectorMode = "dropdown" | "toolbar";
+
+/**
+ * Component to format and highlight code like HTML, CSS, Typescript...
+ * Can be used to display code examples
+ */
+@Component({
+	selector: "stark-language-selector",
+	templateUrl: "./language-selector.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkLanguageSelectorComponent implements OnInit, OnDestroy {
+	/**
+	 * Adds class="stark-language-selector" attribute on the host component
+	 */
+	@HostBinding("class")
+	public class: string = componentName;
+
+	/**
+	 * HTML id of action bar component.
+	 */
+	@Input()
+	public languageSelectorId: string = "";
+
+	/**
+	 * In which 'mode' the language selector is displayed: dropdown / toolbar
+	 */
+	@Input()
+	public mode: StarkLanguageSelectorMode;
+
+	/**
+	 * The currently selected language
+	 */
+	@Input()
+	public selectedLanguage: string;
+
+	/**
+	 * A reference to the sessionService subscription, needed to unsubscribe upon destroy.
+	 */
+	private sessionServiceSubscription: Subscription;
+
+	/**
+	 * Class constructor
+	 * @param appMetadata - The Metadata of the application the contains the supportedLanguages
+	 * @param logger - The logger of the application
+	 * @param sessionService - The session service of the application
+	 * @param dateAdapter - Needed to change the locale of the DateAdapter
+	 */
+	public constructor(
+		@Inject(STARK_APP_METADATA) public appMetadata: StarkApplicationMetadata,
+		@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService,
+		@Inject(STARK_SESSION_SERVICE) public sessionService: StarkSessionService,
+		public dateAdapter: DateAdapter<any>
+	) {}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnInit(): void {
+		this.logger.debug(componentName + ": controller initialized");
+
+		this.sessionServiceSubscription = this.sessionService
+			.getCurrentLanguage()
+			.subscribe(
+				(language: string) => this.selectedLanguage = language,
+				() => this.logger.error(componentName + ": an error occurred getting the current language.")
+			);
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnDestroy(): void {
+		if (this.sessionServiceSubscription) {
+			this.sessionServiceSubscription.unsubscribe();
+		}
+	}
+
+	/**
+	 * Change the current language based on the selection made by the user
+	 */
+	public changeLanguage(language: string): void {
+		
+		if (this.selectedLanguage !== language) {
+			this.logger.debug(componentName + ": setting session current language => " + language);
+			this.selectedLanguage = language;
+			this.sessionService.setCurrentLanguage(this.selectedLanguage);
+			
+			const languageIndex: number = this.appMetadata.supportedLanguages.findIndex((starkLanguage: StarkLanguage) => {
+				return starkLanguage.code === this.selectedLanguage;
+			});
+
+			if (languageIndex >= 0) {
+				const locale: string = this.appMetadata.supportedLanguages[languageIndex].isoCode;
```

Comment:
I'm wondering why we don't do the same check before calling `this.sessionService.setCurrentLanguage(this.selectedLanguage);`, would that make sense?

Reply:
Discussed with @catlabs, this will be fixed when the dropdown issue #640 is merged.